### PR TITLE
EOS-14137: Removed fatal signal handler registration.

### PR DESCRIPTION
**Pull request by DPG17**
Signed-off-by: ‘Dattaprasad <dattaprasad.govekar@seagate.com>

### CODE REVIEW

**Comment by DPG17** [comment]
Per discussion with @knrajnambiar76, removed the registration of fatal handler with OS.
This handler is handling fatal events (SIGABRT, SIGKILL, SIGFPE, SIGSEGV)

There is no need to handle: SIGKILL (as OS will never pass this signal to the process). Rest all fata signals, with their default handler would not require any further custom handler.

**Comment by nileshgovande** [comment]
@knrajnambiar76  and @DPG17 Why did we have it registered originally then?

**Review by amitwac2608 — APPROVED** [review]

File: server/s3server.cc [hunk 1]
```diff
@@ -751,7 +751,6 @@ int main(int argc, char **argv) {
   S3Daemonize s3daemon;
   set_fatal_handler_exit();
   s3daemon.daemonize();
-  s3daemon.register_signals();
```
Comment (knrajnambiar76) [inline server/s3server.cc:754]: you can keep this commented of in #if 0...  since other code is still in place

**Review by knrajnambiar76 — APPROVED** [review]

**Comment by DPG17** [comment]
PR jenkins build: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/112 is triggered.

File: server/s3server.cc (hunk 1 above)
Comment (DPG17) [reply server/s3server.cc:754]: ok.

**Comment by DPG17** [comment]
Jenkins PR build: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server/114/ is triggered.

**Comment by nileshgovande** [comment]
jenkins passed

**Pull request merged**